Common/Interfaces/IDataCacheProvider.cs

```diff
@@ -43,4 +43,33 @@ public interface IDataCacheProvider : IDisposable
         /// <param name="data">The data to cache as a byte array</param>
         void Store(string key, byte[] data);
     }
+
+    // TODO: Better place for this?
+    public static class DataCacheProviderExtensions {
+        /// <summary>
+        /// Helper to separate filename and entry from a given key
+        /// </summary>
+        /// <param name="key">The key to parse</param>
+        /// <param name="fileName">File name extracted</param>
+        /// <param name="entryName">Entry name extracted</param>
+        public static void ParseKey(string key, out string fileName, out string entryName)
```

Comment:
Suggest moving this to `LeanData` he is the one generating the file and entry names after all